lib/elixir/lib/macro.ex

```diff
@@ -478,6 +481,7 @@ defmodule Macro do
 
   """
   @doc since: "1.14.0"
+  @spec path(t, (term -> as_boolean(term))) :: [output] | nil
```

Comment:
I think it should be:
```suggestion
  @spec path(t, (t -> as_boolean(term))) :: [t] | nil
```